src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+        }
+        play->msgCtx.msgMode = 0x44;
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3E9DC.s")
+s32 func_80A3EBFC(EnTest3* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_CLOSING) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EA30.s")
+s32 func_80A3EC30(EnTest3* this, PlayState* play) {
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EAC4.s")
+s32 func_80A3EC44(EnTest3* this, PlayState* play) {
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_CHOICE) && Message_ShouldAdvance(play)) {
+        if (play->msgCtx.choiceIndex != 0) {
+            func_8019F230();
+        } else {
+            func_8019F208();
+        }
+        if (play->msgCtx.choiceIndex != 0) {
+            return 1;
+        } else {
+            s32 ret = this->unk_D78->unk_1 + 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EAF8.s")
+            return ret;
+        }
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EB8C.s")
+s32 func_80A3ECEC(EnTest3* this, PlayState* play) {
+    return D_80A4169C[this->unk_D78->unk_0](this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EBFC.s")
+s32 func_80A3ED24(EnTest3* this, PlayState* play) {
+    s32 temp = D_80A416C0[this->unk_D78->unk_0](this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EC30.s")
+    if (temp != 0) {
+        this->unk_D78 = &this->unk_D78[temp];
+        return func_80A3ECEC(this, play);
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EC44.s")
+void EnTest3_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnTest3* this = THIS;
+    Camera* subCam;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3ECEC.s")
+    if (D_80A41D24) {
+        Actor_MarkForDeath(&this->player.actor);
+        return;
+    }
+    D_80A41D24 = true;
+    this->player.actor.room = -1;
+    this->player.unk_A86 = -1;
+    this->player.transformation = PLAYER_FORM_HUMAN;
+    this->player.ageProperties = &sAgeProperties;
+    this->player.itemActionParam = PLAYER_AP_NONE;
+    this->player.heldItemId = ITEM_OCARINA;
+    Player_SetModelGroup(&this->player, 3);
+    play->playerInit(&this->player, play, &object_test3_Skel_00F7EC);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[0], EFFECT_BLURE2, 0, 0, &sBlureInit);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[1], EFFECT_BLURE2, 0, 0, &sBlureInit);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[2], EFFECT_TIRE_MARK, 0, 0, &sTireMarkInit);
+    this->player.maskObjectSegment = ZeldaArena_Malloc(0x3800);
+    play->func_18780(&this->player, play);
+    this->unk_D90 = GET_PLAYER(play);
```

Comment:
This looks nameable, although it's a classic "we know what it is, but naming is hard". `actualPlayer`?

Reply:
I figured just hold off on naming until someone does a real documentation pass.